src/new/new.action.ts

```diff
@@ -0,0 +1,171 @@
+    })) as { template: string };
+
+    providedTemplate = result.template;
+  }
+
+  if (!newProjectNameRegEx.test(providedTemplate)) {
+    console.error(`Error: Template name ${providedTemplate} is invalid.`);
+    process.exit(1);
+  }
+
+  return providedTemplate;
+}
+
+async function getDescription(providedDescription?: string): Promise<string> {
+  if (typeof providedDescription === "string") {
+    return providedDescription;
+  }
+
+  const result = (await prompt({
+    type: "input",
+    name: "description",
+    message: "Project description",
+    initial: "An unique and awesome application that runs on Golem Network",
+  })) as { description: string };
+
+  return result.description;
+}
+
+type PackageJsonBasic = {
+  name: string;
+  description: string;
+  version: string;
+  author?: string;
+};
+
+async function updatePackageJson(projectPath: string, data: PackageJsonBasic): Promise<void> {
+  const packageJson = join(projectPath, "package.json");
+  let input: string;
+  let json: PackageJsonBasic;
+
+  try {
+    input = await readFile(packageJson, "utf8");
+  } catch (e) {
+    console.error(`Error: Failed to read ${packageJson}: ${e}`);
+    process.exit(1);
+  }
+
+  try {
+    json = JSON.parse(input);
+  } catch (e) {
+    console.error(`Error: Failed to parse ${packageJson}: ${e}`);
+    process.exit(1);
+  }
+
+  json.name = data.name;
+  json.description = data.description;
+  json.version = data.version;
+  if (data.author) {
+    json.author = data.author;
+  }
+
+  try {
+    await writeFile(packageJson, JSON.stringify(json, null, 2));
+  } catch (e) {
+    console.error(`Error: Failed to write ${packageJson}: ${e}`);
+    process.exit(1);
+  }
+}
+
+export async function newAction(providedName: string, options: NewOptions) {
+  const name = await getName(providedName);
+  const projectPath = options.path ?? join(process.cwd(), name);
+  if (existsSync(projectPath)) {
+    console.error(`Error: ${projectPath} already exists.`);
+    process.exit(1);
+  }
+
+  const template = await getTemplate(options.template);
+  const templatePath = join(__dirname, "../../data/project-templates", template);
+  if (!existsSync(templatePath)) {
+    console.error(`Error: Template ${template} not found.`);
+    process.exit(1);
+  }
+
+  const description = await getDescription(options.description);
+  const version = await getVersion(options.version);
+  const author = options.author;
+
+  console.log(`Creating a new Golem app in ${projectPath}.`);
+
+  try {
+    await cp(templatePath, projectPath, { recursive: true });
+  } catch (e) {
+    console.error(`Error: Failed to copy template files: ${e}`);
+    process.exit(1);
+  }
+
+  // Update package.json
+  await updatePackageJson(projectPath, {
+    name,
+    description,
+    version,
+    author,
+  });
+
+  // TODO: Consider running npm install (or yarn, or whatever).
+
+  console.log(`Project created successfully in ${projectPath}.`);
+
+  if (!process.env.YAGNA_APPKEY) {
```

Comment:
to consider whether to create a prompt for APPKEY and place it in .env

Reply:
to be discussed with Grzesiek